util/stringutil/stringutil.go

```diff
@@ -0,0 +1,21 @@
+package stringutil
+
+import (
+	"strconv"
+	"strings"
+)
+
+func StrToInt8Slice(str string) []int8 {
+	var r []int8
+
+	if len(str) > 0 {
+		strSlice := strings.Split(str, ",")
+		for _, s := range strSlice {
+			if v, err := strconv.ParseInt(s, 10, 8); err == nil {
+				r = append(r, int8(v))
+			}
```

Comment:
I was split on this but, in the end I decided to just not append to the array when an error is found. According to its documentation, `ParseInt` returns two types of errors:
1. ErrSyntax: `"If s is empty or contains invalid digits"` (`s` is the input string). In this case, the return value `v` is zero.
1. ErrRange: `"if the value corresponding to s cannot be represented by a signed integer of the given size"`. In this case `v` is `"the maximum magnitude integer of the appropriate bitSize and sign"`.

Didn't think either 0, nor a truncated value of `v` could be useful. I realize the downside of this approach is to be unable to inform the caller that a number in its array couldn't be interpreted as an `int8` but, do we want this? If so, what whould the approach be in case one of the quantities is wrong and the others aren't?

| input | output |
| -- | ------ |
| "1,2" | {1,2}, nil error |
| "1,malformed,3" | {1,3}, ErrSyntax? Or empty array and ErrSyntax? |
| "malformed,malformed" | Empty array and ErrSyntax for sure |
| "12341234123412341234" | {int8max}, ErrRange |

Thoughts?

Reply:
Hm I see two schools of thought. Given that this function is written as a general utility function not specific to GPP, I probably wouldn't swallow the error. Perhaps returning a slice of errors along with any section IDs that were parsed successfully would suffice. The calling code could then decide how to handle the error.

Alternatively, you could just keep what you have here and update the docs comments to point out this behavior.

Let's see what others think.